lib/engine.js

```diff
@@ -38,13 +38,20 @@ Engine = function(initializer) {
   engine = this;
   isStarted = false;
   engine.entityList = Lill.attach({});
-  engine.addEntity = function(components) {
-    var entity;
-    entity = Entity(components);
+  engine.addEntity = function(entity) {
+    if (!entity) {
+      entity = new Entity;
```

Comment:
Not so sure about this, I would expect such behavior rather from `buildEntity`, but `addEntity` should only add the existing one in my opinion.